lib/trino-hdfs/src/main/java/io/trino/hdfs/gcs/HiveGcsConfig.java

```diff
@@ -48,4 +53,13 @@ public HiveGcsConfig setJsonKeyFilePath(String jsonKeyFilePath)
         this.jsonKeyFilePath = jsonKeyFilePath;
         return this;
     }
+
+    public void validate()
+    {
+        // This cannot be normal validation, as it would make it impossible to write TestHiveGcsConfig.testExplicitPropertyMappings
+
+        if (useGcsAccessToken) {
+            checkState(jsonKeyFilePath == null, "Cannot specify 'hive.gcs.json-key-file-path' when 'hive.gcs.use-access-token' is set");
```

Comment:
what about the other way around. 
if `jsonKeyFilePath` is null is it expected that `useGcsAccessToken` is true?

Reply:
they are mutually exclusive, and i don't think it matters which way around i write it?